### DISCUSSION TITLE
Keybinding discussion

### DISCUSSION THREAD

**Discussion by h3nill** [zellij-org/zellij#146] [General]
Does your mind too hurt remembering tmux's keybindings? Here is your chance to give your opinions on it. May it be:
* Prefix/leader key to use
* Mnemonics around which key makes sense for particular action
* anything else? Let us know

**Comment by imsnif** [comment]
With the new status bar, I think we have a great opportunity to rethink keybindings. An idea I had:

Currently we have command mode, which we can access through `ctrl-g`. The problem is that it's very cluttered and does a lot of things. How about grouping some commands we have in it in different modes and having (mostly) the same keys in each mode doing different things depending on the mode.

Here are my thoughts:

Resize mode: `<Ctrl-g r>`
`<arrow-keys/hjkl/emacs-movements>` => resizes current pane

Pane mode: `<Ctrl-g p>`
`<arrow-keys/hjkl/emacs-movements>` => move focus to pane in that direction
`p` => circular focus change (like `<ctrl-g p>` now)
`n` => new pane (find best location)
`d` => split focused pane down (aka horizontal split)
`r` => split current pane right (aka vertical split)
`x` => close focused pane

Tab mode: `<Ctrl-g t>`
`<arrow-keys/hjkl/emacs-movements>` => move focus next/previous tab
`p` => circular focus tab change
`<tab-index>` => move focus to tab index
`n` => new tab
`x` => close current tab

Scroll mode: `<Ctrl-g s>`
In this mode the user should have the ability to move around the pane buffer (both scrolling and moving the cursor around, ideally with both vim/emacs keys and shortcuts)
The user should also be able to mark text and copy it into their OS buffer (both in the vim/emacs ways and "simply" with shift+arrows or in the future with the mouse).

`q` should be a special key that works in all modes and quits
`ESC` always brings the user back to normal mode

At a later stage, I think we can allow users to define macros, so that instead of doing eg. `<ctrl-g r h>` to resize a pane left, they could map it to ctrl-h directly for example. But I think it's important to have to configure these, otherwise we will for sure override some os/terminal shortcuts and have to get into that whole mess :)

**Comment by SaintFenix** [comment]
I really enjoy the current defaults in Zellij, and think that they are sane and quite easy to remember, especially compared to older Multi-plexers.

I've put in PR #362 to address possible future tmux compatible binds for people switching over.

I honestly will be installing Zellij on all my machines once session detaching is finished being implemented, I can't think of a reason I'd prefer tmux as the binds here are just, so much easier to use, and learn about.
